src/DataStructures/SpinWeighted.hpp

```diff
@@ -18,13 +20,113 @@
  * two summands possess the same spin weight, and multiplication (or division)
  * result in a summed (or subtracted) spin weight.
  */
+template <typename T, int Spin, bool is_vector = is_derived_of_vector_impl_v<T>>
+struct SpinWeighted;
+
 template <typename T, int Spin>
-struct SpinWeighted {
+struct SpinWeighted<T, Spin, false> {
   T data;
```

Comment:
Since this is no longer a POD (at least I'm fairly sure it no longer is), we should make the `T data_` private with a `data()` accessor member function.